airsonic-main/src/test/java/org/airsonic/player/service/metadata/MetaDataFactoryTestCase.java

```diff
@@ -0,0 +1,86 @@
+package org.airsonic.player.service.metadata;
+
+import org.airsonic.player.TestCaseUtils;
+import org.airsonic.player.service.SettingsService;
+import org.airsonic.player.util.HomeRule;
+import org.apache.commons.io.FileUtils;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertThat;
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class MetaDataFactoryTestCase {
+
+    @ClassRule
+    public static final SpringClassRule classRule = new SpringClassRule() {
+        HomeRule airsonicRule = new HomeRule() {
+            @Override
+            protected void before() throws Throwable {
+                super.before();
+                String homeParent = TestCaseUtils.airsonicHomePathForTest();
+                System.setProperty("airsonic.home", TestCaseUtils.airsonicHomePathForTest());
+                TestCaseUtils.cleanAirsonicHomeForTest();
+                File dbDirectory = new File(homeParent, "/db");
+                FileUtils.forceMkdir(dbDirectory);
+                org.airsonic.player.util.FileUtils.copyResourcesRecursively(getClass().getResource("/db/pre-liquibase/db"), new File(homeParent));
```

Comment:
I think this is a potential bug for now.
It also exists in LegacyDatabaseStartupTestCase.

pre-liquibase files are libresonic. The default jdbc url will point to airsonic (because of the presence of string 'airsonic' name in the temporary test folder).

Reply:
This isn't correct. See #1285

Reply:
What isn't correct? This test class is from your commit.